pandas/core/arrays/integer.py

```diff
@@ -0,0 +1,564 @@
+import sys
+import warnings
+import copy
+import numpy as np
+
+from pandas.compat import u
+from pandas.core.dtypes.generic import ABCSeries, ABCIndexClass
+from pandas.util._decorators import cache_readonly
+from pandas.compat import set_function_name
+from pandas.api.types import (is_integer, is_scalar, is_float,
+                              is_float_dtype, is_integer_dtype,
+                              is_object_dtype,
+                              is_list_like,
+                              infer_dtype)
+from pandas.core.arrays import ExtensionArray, ExtensionOpsMixin
+from pandas.core.dtypes.base import ExtensionDtype
+from pandas.core.dtypes.dtypes import registry
+from pandas.core.dtypes.missing import isna, notna
+from pandas.core.dtypes.cast import maybe_downcast_to_dtype
+
+from pandas.io.formats.printing import (
+    format_object_summary, format_object_attrs, default_pprint)
+
+
+class IntegerDtype(ExtensionDtype):
+    type = None
+    na_value = np.nan
+
+    @cache_readonly
+    def is_signed_integer(self):
+        return self.kind == 'i'
+
+    @cache_readonly
+    def is_unsigned_integer(self):
+        return self.kind == 'u'
+
+    @cache_readonly
+    def numpy_dtype(self):
+        """ Return an instance of our numpy dtype """
+        return np.dtype(self.type)
+
+    @cache_readonly
+    def kind(self):
+        return self.numpy_dtype.kind
+
+    @classmethod
+    def construct_array_type(cls):
+        """Return the array type associated with this dtype
+
+        Returns
+        -------
+        type
+        """
+        return IntegerArray
+
+    @classmethod
+    def construct_from_string(cls, string):
+        """
+        Construction from a string, raise a TypeError if not
+        possible
+        """
+        if string == cls.name:
+            return cls()
+        raise TypeError("Cannot construct a '{}' from "
+                        "'{}'".format(cls, string))
+
+
+def to_integer_array(values):
+    """
+    Parameters
+    ----------
+    values : 1D list-like
+
+    Returns
+    -------
+    infer and return an integer array
+
+    Raises
+    ------
+    TypeError if incompatible types
+    """
+    values = np.array(values, copy=False)
+    try:
+        dtype = _dtypes[str(values.dtype)]
+    except KeyError:
+        if is_float_dtype(values):
+            return IntegerArray(values)
+
+        raise TypeError("Incompatible dtype for {}".format(values.dtype))
+    return IntegerArray(values, dtype=dtype, copy=False)
+
+
+def coerce_to_array(values, dtype, mask=None, copy=False):
+    """
+    Coerce the input values array to numpy arrays with a mask
+
+    Parameters
+    ----------
+    values : 1D list-like
+    dtype : integer dtype
+    mask : boolean 1D array, optional
+    copy : boolean, default False
+        if True, copy the input
+
+    Returns
+    -------
+    tuple of (values, mask)
+    """
+
+    if isinstance(values, IntegerArray):
```

Comment:
do we need to check the dtype argument in this case as well?

Reply:
done